docs/development/extensions-contrib/gce-extensions.md

```diff
@@ -0,0 +1,86 @@
+  ~ to you under the Apache License, Version 2.0 (the
+  ~ "License"); you may not use this file except in compliance
+  ~ with the License.  You may obtain a copy of the License at
+  ~
+  ~   http://www.apache.org/licenses/LICENSE-2.0
+  ~
+  ~ Unless required by applicable law or agreed to in writing,
+  ~ software distributed under the License is distributed on an
+  ~ "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+  ~ KIND, either express or implied.  See the License for the
+  ~ specific language governing permissions and limitations
+  ~ under the License.
+  -->
+
+
+To use this Apache Druid (incubating) extension, make sure to [include](../../development/extensions.md#loading-extensions) `gce-extensions`.
+
+At the moment, this extension enables only Druid to autoscale instances in GCE.
+
+The extension manages the instances to be created and destroyed through the Managed Instance Groups of GCE —MIG from
+now on— and not dealing with individual machines.  This to enable ans easier configuration, as all the basic config
+is managed at the group level and for ease of human intervention in case of need rescaling the group through the GCE 
+interface and tools.
+
+## Overlord Dynamic Configuration
+
+The Overlord can dynamically change worker behavior.
+
+The JSON object can be submitted to the Overlord via a POST request at:
+
+```
+http://<OVERLORD_IP>:<port>/druid/indexer/v1/worker
+```
+
+Optional Header Parameters for auditing the config change can also be specified.
+
+|Header Param Name| Description | Default |
+|----------|-------------|---------|
+|`X-Druid-Author`| author making the config change|""|
+|`X-Druid-Comment`| comment describing the change being done|""|
+
+A sample worker config spec is shown below:
+
+```json
+{
+  "autoScaler": {
+    "envConfig" : {
+      "targetWorkers" : 1,
+      "projectId" : "super-project",
+      "zoneName" : "us-central-1",
+      "managedInstanceGroupName" : "druid-mig"
+    },
+    "maxNumWorkers" : 4,
+    "minNumWorkers" : 2,
+    "type" : "gce"
+  }
+}
+```
+
+The configuration of the autoscaler is quite simple and it it made of two levels only.
+
+The external level specifies the `type`—always `gce` in this case— and two numeric values,
+the `maxNumWorkers` and `minNumWorkers` used to define the boundwaries in between which the
+number of instances must be at any time.
+
+The internal level is the `envConfig` use to specify
+
+- The `targetWorkers` used to specify how many workers will be spawned at each 
+request to provision more workers.  This is safe to be left to `1`
+- The `projectId` used to specify the name of the project in which the MIG resides
+- The `zoneName` used to identify in which zone of the worlds the MIG is
+- The `managedInstanceGroupName` used to specify the MIG cotaining the instances created or 
+removed
+
+Please refer to the Overlord Dynamic Configuration section in the main [documentation](../../configuration/index.md)
```

Comment:
shall we add a note about the service account? (required scopes and which is the environment variable to use and what does it do if it is undefined -- i guess it uses the instance default service account)

Reply:
yes!

Reply:
... :-)